junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/HierarchicalTestExecutor.java

```diff
@@ -93,13 +115,26 @@ private void execute(TestDescriptor testDescriptor, C parentContext, ExecutionTr
 				// @formatter:on
 			}
 			finally {
-				node.after(context);
+				try {
+					node.after(context);
+				}
+				finally {
+					node.cleanUp(context);
```

Comment:
I am not happy with this `try-finally` block inside a `finally` block.

Additionally, when you launch a single demo (`HeavyweightAlphaTests` or `HeavyweightBetaTests`) the assertion exception generated in the `close()` method in class `Heavyweight.ResourceValue` is "swallowed and suppressed". Could point to an IDE(A) issue, though.

Reply:
@marcphilipp You xmas present... :)